File: PAT/basic_level/1069_微博转发抽奖/raffle_for_weibo_followers.cpp
#include <iostream>
#include <string>
#include <set>
using namespace std;

set<string> win;

int main(){
	string nickname;
	int m, step, s, start, i;
	cin>>m>>step>>s;
	start = s;
	
	for( i = 1; i <= m; ++i ){
		cin>>nickname;
		if( i == start ){
			if( win.find(nickname) == win.end() ){
				cout<<nickname<<'\n';
				win.insert(nickname);
				start += step;
			}else{
				start++;
			}
		}
	}
	if( s > m ){
		cout<<"Keep going...\n";
		return 0;
	}
	return 0;
}

File: PAT/basic_level/1069_微博转发抽奖/raffle_for_weibo_followers.py
#!/usr/bin/env python3

s = input().split(' ')
totalNumber, step, firstWinner = map(int, s)
repostList = ['']
winners = set()

for i in range(totalNumber):
	repostList.append( input() )

if firstWinner > totalNumber:
	print('Keep going...')
else:
	i = firstWinner
	while i<=totalNumber:
		candidate = repostList[i]
		if candidate in winners:
			while i<=totalNumber:
				nextCandidate = repostList[i]
				if nextCandidate not in winners:
					print(nextCandidate)
					winners.add(nextCandidate)
					break
				i += 1
		else:
			print(candidate)
			winners.add(candidate)
		i += step
